### ISSUE TITLE
Cannot Establish Streaming Connection to FedRAMP Instance

### ISSUE THREAD

**Issue by drewpc** [launchdarkly/erlang-server-sdk#127]
**Describe the bug**
Beginning around May 7, 2024, the Erlang Server SDK stopped connecting to the streaming FedRAMP server at https://stream.launchdarkly.us.  This seems to be due to a cipher mismatch while negotiating the TLS connection.  I believe what is happening is that the server now supports both TLS v1.2 and v1.3, but the Erlang Server SDK attempts to negotiate a TLS v1.3 connection with only ciphers for TLS v1.2.  Wireshark packet captures show the client initiating a TLS connection and the server immediately closing with an Alert "close notify" packet.

The issue appears to be in the code that filters ciphers based on key exchange algorithms.  This code filters out any ciphers for only TLS v1.2:
https://github.com/launchdarkly/erlang-server-sdk/blob/0b2efee3141a74370710b280d50db1cac0274225/src/ldclient_config.erl#L398-L420

If LaunchDarkly's intent is to only support TLS v1.2, then `tls_base_options/0` must be updated to restrict the connection to TLS v1.2 only.  If the intent is to support both TLS v1.2 and v1.3, then ciphers for both TLS v1.2 and v1.3 must be included in the list.  Specific information can be found on the [Erlang Hardening SSL page](https://erlef.github.io/security-wg/secure_coding_and_deployment_hardening/ssl.html#selecting-protocol-versions-and-ciphers).

**To reproduce**
The following code samples are implemented in Elixir using `iex -S mix`.

This fails:
```elixir
opts = %{
  :base_uri => 'https://sdk.launchdarkly.us',
  :events_uri => 'https://events.launchdarkly.us',
  :http_options => %{
    :tls_options => :ldclient_config.tls_basic_options()
  },
  :stream_uri => 'https://stream.launchdarkly.us'
}

:ldclient.start_instance(ld_sdk_key, :default, opts)
```

This succeeds by explicitly only allowing TLS v1.2 connections to the server:
```elixir
opts = %{
  :base_uri => 'https://sdk.launchdarkly.us',
  :events_uri => 'https://events.launchdarkly.us',
  :http_options => %{
    :tls_options => :ldclient_config.tls_basic_options() ++ [
      versions: [:"tlsv1.2"]
    ]
  },
  :stream_uri => 'https://stream.launchdarkly.us'
}

:ldclient.start_instance(ld_sdk_key, :default, opts)
```

**Expected behavior**
Prior to May 7, 2024, these connections to the stream server succeeded and all flag data was retrieved.

**Logs**
```log
18:49:12.887 [notice] Starting streaming update server for :default
18:49:12.891 [notice] Starting streaming connection to URL: ~c"https://stream.launchdarkly.us/all"
18:49:13.238 [warning] Error establishing streaming connection (:temporary): ~c"Could not open connection to host", will retry in 546 ms
18:49:13.786 [notice] Reconnecting streaming connection...
```
![image](https://github.com/launchdarkly/erlang-server-sdk/assets/1407905/db2c94c0-1199-4d98-ba67-9c2bdb9c333e)
**SDK version**
version 3.2.0

**Language version, developer tools**
erlang 25.3.2
elixir 1.15.4-otp-25

**OS/platform**
macOS 14.4.1 and Ubuntu 20.04

**Comment by cwaldren-ld** [comment]
Hi @drewpc , appreciate you reporting this. 

Is this currently impacting your operations, or are you in a testing/development phase with the Erlang SDK?

**Comment by drewpc** [comment]
The workaround I suggested above is sufficient for us that it won't impact operations when we move into production in the next few weeks.

**Comment by kinyoklion** [comment]
@drewpc Version 3.3.0 of the SDK has been released and supports TLS 1.3 on OTP 23 and greater. When on an OTP version older than 23 it will limit the connection to TLS 1.2.

Thank you!
Ryan

**Comment by drewpc** [comment]
thank you so much for the quick response!
